client_test.go

```diff
@@ -312,6 +311,85 @@ func TestClientFast(t *testing.T) {
 	CheckNumMessages(t, rabbitC, queueName, 0)
 }
 
+func TestPanicHandlingInConsumer(t *testing.T) {
+	t.Parallel()
+	rabbitC, url := runRabbitContainer(t)
+	defer func() { require.NoError(t, rabbitC.Terminate(context.Background())) }()
+
+	const dlqQueueName = "queue-dlq"
+	const dlqExchangeName = "exchange-dlq"
+	const queueName = "queue"
+	require.NoError(t, declareQueue(t, rabbitC, dlqQueueName))
+	require.NoError(t, declareExchange(t, rabbitC, dlqExchangeName, "fanout"))
+	require.NoError(t, declareBinding(t, rabbitC, dlqExchangeName, dlqQueueName))
+	require.NoError(t, declareQueueWithDlq(t, rabbitC, queueName, dlqExchangeName))
+
+	num := 10
+	received := make(chan bool, num) // this channel gets "released" on message delivery
+	// Consumer
+	qi, err := NewQueue(url, queueName, 1, true, false, nil)
+	require.NoError(t, err)
+	defer qi.Close()
+
+	const numberForWhichPublisherPanic = 4
+	rec := func(m amqp.Delivery) *amqp.Publishing {
+		if m.CorrelationId == strconv.Itoa(numberForWhichPublisherPanic) {
+			panic("panicking !!!!!")
+		}
+
+		assert.NoError(t, m.Ack(false))
+		received <- true
+		return nil
+	}
+	assert.NoError(t, qi.AddReceiver(rec))
```

Comment:
is it possible to check it multiple receivers ?

Reply:
yes, but for what? Test for one receiver is good, because we can test that consumer is able to process all messages, even if one of them causes receiver to panic